[feature/#719] Add Benchmark Module and baseline profile generator

l2hyunwoo opened the pull request:
## Issue
- close #719 

## Overview (Required)
- Add Benchmark Module and baseline profile generator
- There's an [issue](https://github.com/DroidKaigi/conference-app-2023/issues/719#issuecomment-1697356674)

takahirom commented on app-android/src/devRelease/generated/baselineProfiles/baseline-prof.txt:
Sorry I'm going to create this by myself without committing so could you remove this?

l2hyunwoo replied:
It's deleted

takahirom commented:
@l2hyunwoo Do you have any idea why this CI takes 1 hour? 👀

l2hyunwoo commented:
@takahirom It's hard to find the reason why it takes much longer than others..... Reasonable reason that I can think is there's no build cache related to baseline-profile. So if build cache is created, build time will restore I suppose.

+) One more I think is there're procedures that related to prod variant, kind of strange things....(I thought it just assembleDev..) It could increase build time much more
<img width="805" alt="스크린샷 2023-08-31 오후 11 13 32" src="https://github.com/DroidKaigi/conference-app-2023/assets/54518925/952504fa-afa3-4b19-8329-d2b3dd48ba74">

takahirom commented:
@l2hyunwoo 
Well, you can try to use increased build time for the GitHub Action.  🙏

l2hyunwoo committed:
[feature/#719] Add baselineprofile module


[feature/#719] Add baseline-prof when variant is dev


[feature/#719] Delete baseline-prof


[feature/#719] Fix local build

l2hyunwoo committed:
Detekt auto correct
